server/services/messageServices.js

```diff
@@ -175,35 +178,38 @@ export default class MessageService {
         messageId: newMessage.id,
       });
     }
+    this.messages.push(newMessage);
     return newMessage;
   }
 
   getMessageById(id) {
-    const message = this.AllMessage()[id - 1];
+    const foundMessage = this.AllMessage().find(element => element.id === Number(id));
     if (!id) {
       return 'error';
     }
-    if (!message) {
+    if (!foundMessage) {
       return 'error';
     }
-    return message;
+    return foundMessage;
   }
 
   deleteMessage(id) {
-    const message = this.AllMessage()[id - 1];
+    const foundMessage = this.messages.find(element => element.id === Number(id));
     if (!id) {
```

Comment:
This check for the `id` should also come before you search for the message

Reply:
Done